tests/sagemaker/mock/__init__.py

```diff
@@ -514,13 +516,20 @@ def list_models(self):
             summaries.append(summary)
         return summaries
 
-    def list_tags(self, resource_arn, region_name):  # pylint: disable=unused-argument
+    def list_tags(
+        self, resource_arn, region_name, resource_type
+    ):  # pylint: disable=unused-argument
         """
         Modifies backend state during calls to the SageMaker "ListTags" API
         https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_ListTags.html
         """
-        model = next(model for model in self.models.values() if model.arn == resource_arn)
-        return model.resource.tags
+        resource_values = getattr(self, resource_type).values()
+        sagemaker_resource = next(
```

Comment:
Usually if we need to break inline code to multiple lines, it indicates we should replace the inline style with the normal for loop for readability. could you help refactor? thanks!